sycl/plugins/hip/CMakeLists.txt

```diff
@@ -13,7 +13,26 @@ if(NOT EXISTS "${SYCL_BUILD_PI_HIP_ROCM_DIR}")
 endif()
 
 # Set HIP include dirs
-set(HIP_HEADERS "${SYCL_BUILD_PI_HIP_ROCM_DIR}/hip/include;${SYCL_BUILD_PI_HIP_ROCM_DIR}/hsa/include")
+if(NOT DEFINED SYCL_BUILD_PI_HIP_INCLUDE_DIR)
+  set(SYCL_BUILD_PI_HIP_INCLUDE_DIR "${SYCL_BUILD_PI_HIP_ROCM_DIR}/hip/include")
+endif()
```

Comment:
I think this works, although I think we should keep the cache variables as it is nicer for CMake GUIs and such, just have the default value empty, that way it's there but if the user hasn't set it we can tell and use the other variable instead, something like this:

```cmake
if("${SYCL_BUILD_PI_HIP_INCLUDE_DIR}" STREQUAL "")
  set(PI_HIP_INCLUDE_DIR "${SYCL_BUILD_PI_HIP_INCLUDE_DIR}")
else()
  set(PI_HIP_INCLUDE_DIR "${SYCL_BUILD_PI_HIP_ROCM_DIR}/hip/include")
endif()
```

Also could you update the check just above this as well, if a user sets the specific variables but doesn't set `SYCL_BUILD_PI_HIP_ROCM_DIR` we shouldn't throw an error.

Reply:
Ahh, nevermind, I think I understand what you are saying above. The goal isn't to have changes to `SYCL_BUILD_PI_HIP_ROCM_DIR` automatically set the `SYCL_BUILD_PI_HIP_INCLUDE_DIR` cached variables. Just leave them empty and if the user changes them, updating those cached variables are up to them. Let me know if this is the case. Then I will go back to cached variables and mark them as advanced with default empty values. And set up conditional checks as you described above.